### ISSUE TITLE
CoffeeScript %%= becomes %% =

### ISSUE THREAD

**Issue by pzp1997** [Glavin001/atom-beautify#283]
The true mathematical modulo assignment operator (%%=) gets "corrected" to %% =, which is incorrect and causes an error (unexpected =). Here's my debugging information:

# Atom Beautify - Debugging information

The following debugging information was generated by `Atom Beautify` on `Mon Apr 20 2015 12:42:00 GMT-0400 (EDT)`.

---

**Platform**: darwin

## Versions

**Atom Version**: 0.192.0

**Atom Beautify Version**: 0.24.0

## Original file to be beautified

**Original File Path**: `~/Desktop/Theremin/scripts/theremin.coffee`

**Original File Grammar**: CoffeeScript

**Original File Contents**: 
```CoffeeScript
$(document).ready ->
  AudioContext = window.AudioContext or window.webkitAudioContext
  audioCtx = new AudioContext()

  oscillator = audioCtx.createOscillator()
  gainNode = audioCtx.createGain()

  oscillator.connect gainNode
  gainNode.connect audioCtx.destination

  oscillator.type = 'square'
  oscillator.frequency.value = 2000
  oscillator.start 0

  gainNode.gain.value = 0.5

  $(document).keydown (event) ->
    switch event.which
      when 38 # Up arrow
        oscillator.frequency.value += 100
      when 40 # Down arrow
        oscillator.frequency.value -= 100

    oscillator.frequency.value %%= 4000

```

## Beautification options

**Editor Options**: 
Options from Atom Editor settings
```json
{
    "indent_size": 2,
    "indent_char": " ",
    "indent_with_tabs": false
}
```

**Config Options**: 
Options from Atom Beautify package settings
```json
{
    "js": {
        "space_after_anon_function": true,
        "indent_size": 2,
        "indent_char": " ",
        "indent_level": 0,
        "indent_with_tabs": false,
        "preserve_newlines": true,
        "max_preserve_newlines": 10,
        "space_in_paren": false,
        "jslint_happy": false,
        "brace_style": "collapse",
        "break_chained_methods": false,
        "keep_array_indentation": false,
        "keep_function_indentation": false,
        "space_before_conditional": true,
        "eval_code": false,
        "unescape_strings": false,
        "wrap_line_length": 0,
        "end_with_newline": false
    },
    "css": {
        "indent_size": 2,
        "indent_char": " ",
        "selector_separator_newline": false,
        "newline_between_rules": false,
        "preserve_newlines": false
    },
    "html": {
        "htmlbeautifier_path": "",
        "indent_inner_html": false,
        "indent_size": 2,
        "indent_char": " ",
        "brace_style": "collapse",
        "indent_scripts": "normal",
        "wrap_line_length": 250,
        "wrap_attributes": "auto",
        "wrap_attributes_indent_size": 2,
        "preserve_newlines": true,
        "max_preserve_newlines": 10,
        "unformatted": [
            "a",
            "sub",
            "sup",
            "b",
            "i",
            "u"
        ],
        "end_with_newline": false
    },
    "sql": {
        "indent_size": 2,
        "keywords": "upper",
        "identifiers": "lower",
        "sqlformat_path": ""
    },
    "markdown": {
        "pandoc_path": "",
        "yaml_front_matter": true
    },
    "perl": {
        "perltidy_path": "perltidy",
        "perltidy_profile": ""
    },
    "php": {
        "cs_fixer_path": "",
        "fixers": "",
        "level": ""
    },
    "python": {
        "autopep8_path": "",
        "max_line_length": 79,
        "indent_size": 2,
        "ignore": [
            "E24"
        ]
    },
    "ruby": {
        "rbeautify_path": ""
    },
    "c": {
        "uncrustifyPath": "",
        "configPath": ""
    },
    "cpp": {
        "uncrustifyPath": "",
        "configPath": ""
    },
    "objectivec": {
        "uncrustifyPath": "",
        "configPath": ""
    },
    "cs": {
        "uncrustifyPath": "",
        "configPath": ""
    },
    "d": {
        "uncrustifyPath": "",
        "configPath": ""
    },
    "java": {
        "uncrustifyPath": "",
        "configPath": ""
    },
    "pawn": {
        "uncrustifyPath": "",
        "configPath": ""
    },
    "vala": {
        "uncrustifyPath": "",
        "configPath": ""
    }
}
```

**Home Options**: 
Options from `~/.jsbeautifyrc`
```json
{}
```

**EditorConfig Options**: 
Options from [EditorConfig](http://editorconfig.org/) file
```json
{}
```

**Project Options**: 
Options from `.jsbeautifyrc` files starting from directory `~/Desktop/Theremin/scripts` and going up to root
```json
[
    {},
    {},
    {},
    {},
    {}
]
```

## Logs

**Error logs**: *Not yet supported*

**Comment by Glavin001** [comment]
This is a bug with Coffee-formatter: https://github.com/Glavin001/Coffee-Formatter
Unfortunatly, the original author ( https://github.com/derekchiang/Coffee-Formatter ) no longer maintains this library. I have been looking for another.

I think this could be solved by adding `%` to the list of `TWO_SPACE_OPERATORS` ( see https://github.com/Glavin001/Coffee-Formatter/blob/065cba041571511480d5f79ee25ca61393b17136/src/formatter.litcoffee#constants ). Similar fix as https://github.com/Glavin001/Coffee-Formatter/commit/065cba041571511480d5f79ee25ca61393b17136

**Comment by Glavin001** [comment]
I've created an issue: https://github.com/Glavin001/Coffee-Formatter/issues/3
Will take a look when I have some more time. Thanks for posting this issue!

**Comment by Glavin001** [comment]
Looking into switching to using [coffee-fmt](https://github.com/sterpe/coffee-fmt) by @sterpe. See #192

**Comment by Glavin001** [comment]
I'ved patched Coffee-Formatter and released 0.1.2: https://www.npmjs.com/package/coffee-formatter

On Atom Beautify's next release, v0.26.0, you will have the latest Coffee-Formatter with this fix.

**Comment by Glavin001** [comment]
Published to v0.26.0

Coffee-Formatter has been fixed for this bug, as well as coffee-fmt is now available (and default) and works even better! See #192
